src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -166,6 +166,12 @@ public class InstancesController : ControllerBase
                 {
                     return BadRequest("InstanceOwnerPartyId must be defined.");
                 }
+
+                // filter out hard deleted instances if user is requesting instances          
+                if (!Request.QueryString.HasValue || !Request.QueryString.Value.Contains("status.isHardDeleted"))
+                {
+                    Request.QueryString.Add("status.isHardDeleted", "false");
```

Comment:
I would like to avoid us using the Request property as anything other than data source for logic. We should in my opinion never write to it. Could we move this logic a bit further down and add the information directly  to `queryParams` instead?